tests/configuration.py

```diff
@@ -0,0 +1,105 @@
+        for symbol in dir(settings):
+            if symbol.startswith('EMARSYS_'):
+                delattr(settings, symbol)
+
+        util.validate_configuration()
+
+        expected_messages = set([
+            'no EMARSYS_ACCOUNT setting set',
+            'no EMARSYS_PASSWORD setting set',
+            'no EMARSYS_BASE_URI setting set',
+            'no EMARSYS_EVENTS setting set',
+        ])
+        actual_messages = [message.getMessage()
+                           for message in handler.messages]
+
+        for message in actual_messages:
+            if message not in expected_messages:
+                raise self.failureException("got unexpected log message '{}'"
+                                            .format(message))
+
+            expected_messages.remove(message)
+
+        if expected_messages:
+            raise self.failureException(
+                "expected log message(s) missing: {}"
+                .format(list(expected_messages)))
+
+    @override_settings()
+    def test_configuration_integrity(self):
+        logger = logging.getLogger(util.__file__)
+        handler = MockLoggingHandler(level='WARNING')
+        logger.addHandler(handler)
+
+        settings.EMARSYS_EVENTS = {
+            '': [],
+            'test event 1': [
+                dict(name="Reused Name", arg="reused_arg", model="auth.User"),
+                dict(name="Reused Name", arg="reused_arg", model="auth.User"),
+                dict(name="foobar", arg="", model="auth.User"),
+                dict(name="", arg="invalid arg", model="auth.User"),
+                dict(name="Not Reused Name", arg="not_reused_arg",
+                     model="auth.User"),
+            ],
+            'test event 2': [
+                dict(name="Not Reused Name", arg="not_reused_arg",
+                     model="auth.User"),
+                dict(foobar="abc"),
+            ],
+            'test event 3': [
+                dict(name="User1", arg="user1", model="invalid_app.User"),
+                dict(name="User2", arg="user2", model="auth.InvalidModel"),
+            ],
+
+        }
+
+        util.validate_configuration()
+
+        expected_messages = set([
+            "invalid event name: ''",
+            "reused data name for event 'test event 1': 'Reused Name'",
+            "reused data argument for event 'test event 1': 'reused_arg'",
+            "invalid data argument for event 'test event 1': ''",
+            "invalid data argument for event 'test event 1': 'invalid arg'",
+            "invalid data name for event 'test event 1': ''",
+            "unexpected data key for event 'test event 2': 'foobar'",
+            "missing data key for event 'test event 2': 'name'",
+            "missing data key for event 'test event 2': 'arg'",
+            "missing data key for event 'test event 2': 'model'",
+            "bad model 'invalid_app.User' for event 'test event 3': "
+            + "No installed app with label 'invalid_app'.",
```

Comment:
`+` is unnecessary here. To make it more clear visually that this is actually a string continuation, consider putting the string in parentheses.